connector/src/yang/connector/grpc/telegraf.py

```diff
@@ -0,0 +1,133 @@
+                    config.write(f)
+            else:
+                # set config file path
+                self.config_file = f"{runtime.directory}/transporter.conf"
+                config.read(self.config_file)
+
+                # if the file already exists, only update the port
+                if config.sections():
+                    if '[inputs.cisco_telemetry_mdt]' not in config.sections():
+                        config.add_section('[inputs.cisco_telemetry_mdt]')
+
+                    config.set('[inputs.cisco_telemetry_mdt]', 'transport', '"grpc"')
+                    config.set('[inputs.cisco_telemetry_mdt]', 'service_address', f'":{allocated_port}"')
+
+                    # Don't overwrite first file, stick that in /tmp/
+                    self.config_file = f"{self.config_directory}/transporter.conf"
+                    with open(f"{self.config_directory}/transporter.conf", 'w') as f:
+                        log.info(f"Writing config to {self.config_directory}/transporter.conf")
+                        config.write(f)
+                else:
+                    # generate a default configuration file
+                    with open(self.config_file, 'w') as f:
+                        log.info(f"Creating telegraf config file {self.config_file}")
+                        f.write('')
+
+                    # create default config
+                    # global tags
+                    config.add_section('global_tags')
+                    config.set('global_tags', 'user', r'"${USER}"')
+
+                    # input configuration
+                    config.add_section('[inputs.cisco_telemetry_mdt]')
+                    config.set('[inputs.cisco_telemetry_mdt]', 'transport', '"grpc"')
+                    config.set('[inputs.cisco_telemetry_mdt]', 'service_address', f'":{allocated_port}"')
+
+                    # default output config - to file in runtime or user supplied dir
+                    config.add_section('[outputs.file]')
+                    config.set('[outputs.file]', 'files', f'["stdout", "{self.output_file}"]')
+                    config.set('[outputs.file]', 'data_format', '"json"')
+                    config.set('[outputs.file]', 'json_timestamp_units', '"1ms"')
+                    config.set('[outputs.file]', 'rotation_max_size', '"2048MB"')
+                    config.set('[outputs.file]', 'flush_jitter', '"500ms"')
+
+                    # apply config
+                    with open(self.config_file, 'w') as f:
+                        log.info(f"Updating {self.config_file}")
+                        config.write(f)
+
+        # exit context manager to release port
+        # spawn telegraf/pipeline using config
+        self.transport_process = subprocess.Popen(f"telegraf -config '{self.config_file}'", shell=True)
+
+        # log port
+        log.info(f"Telegraf is running as PID {self.transport_process.pid} on port {allocated_port}")
```

Comment:
Please update the device.connections['grpc'] datastructure with the transporter ip and port so the user can access this (in case the configuration is not done automatically)